File: ApiSuvesaPos/Negocio/Logica/PreTomaGeneral.vb
Namespace Logica
    Public Class PreTomaGeneral

        Private db As Datos.Class.PreTomaGeneral

        Sub New()
            Me.db = New Datos.Class.PreTomaGeneral
        End Sub

        Public Function Crear(pretoma As Datos.Models.PreTomaGeneral) As String
            Return Me.db.Crear(pretoma)
        End Function

        Public Function Editar(id As Decimal, pretoma As Datos.Models.PreTomaGeneral) As String
            Dim res As String = Me.db.Editar(id, pretoma)
            If res = "0" Then
                Return "No existe el valor"
            Else
                Return res
            End If
        End Function

        Public Function Buscar(porNombre As Boolean, Filtro As String) As List(Of Datos.Models.PreTomaGeneral)
            Dim datos As New List(Of Datos.Models.PreTomaGeneral)
            datos = db.Buscar(porNombre, Filtro)

            Dim Resultado As New List(Of Datos.Models.PreTomaGeneral)
            For Each toma As Datos.Models.PreTomaGeneral In datos
                Dim preToma As New Datos.Models.PreTomaGeneral
                preToma.IdPreToma = toma.IdPreToma
                preToma.Fecha = toma.Fecha
                preToma.Anulado = toma.Anulado
                preToma.Aplicado = toma.Aplicado
                preToma.Encargado = toma.Encargado

                For Each detalle As Datos.Models.PreTomaGeneralDetalle In Me.db.BuscarDetalle(preToma.IdPreToma)
                    Dim preTomaDetalle As New Datos.Models.PreTomaGeneralDetalle
                    preTomaDetalle.IdPreTomaDetalle = detalle.IdPreTomaDetalle
                    preTomaDetalle.IdPreToma = detalle.IdPreToma
                    preTomaDetalle.Codigo = detalle.Codigo
                    preTomaDetalle.CodArticulo = detalle.CodArticulo
                    preTomaDetalle.Barras = detalle.Barras
                    preTomaDetalle.Descripcion = detalle.Descripcion
                    preTomaDetalle.Costo = detalle.Costo
                    preTomaDetalle.Existencia = detalle.Existencia
                    preTomaDetalle.Toma = detalle.Toma
                    preTomaDetalle.Diferencia = detalle.Diferencia
                    preTomaDetalle.Contado = detalle.Contado

                    preToma.PreTomaGeneralDetalles.Add(preTomaDetalle)
                Next

                Resultado.Add(preToma)
            Next

            Return Resultado
        End Function

    End Class
End Namespace

File: ApiSuvesaPos/Datos/Class/PreTomaGeneral.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Datos.Models;
using Microsoft.EntityFrameworkCore;

namespace Datos.Class
{
	public class PreTomaGeneral
	{
		private SeePOSContext entities;

		public PreTomaGeneral()
		{
			entities = new SeePOSContext();
		}

		public int Crear(Models.PreTomaGeneral pretomageneral) // registro de PreTomaGeneral
		{
			try
			{
				entities.PreTomaGenerals.Add(pretomageneral);
				return entities.SaveChanges();
			}
			catch (Exception ex)
			{
				throw ex;
			}
		}

		public List<Models.PreTomaGeneral> Buscar(bool porNombre, string filtro)  //consultar PreTomaGeneral
		{
			try
			{
				List<Models.PreTomaGeneral> result;
				if (porNombre == true)
				{
					var temp = from c in entities.PreTomaGenerals
							   select c;
					result = temp.ToList<Models.PreTomaGeneral>();
				}
				else
				{
					var temp = from c in entities.PreTomaGenerals
							   select c;
					result = temp.ToList<Models.PreTomaGeneral>();
				}
				if (result.Count > 0)
				{
					return result;
				}
				else
				{
					return result = null;
				}
			}
			catch (Exception ex)
			{
				throw ex;
			}
		}

		public List<Models.PreTomaGeneralDetalle> BuscarDetalle(long Id)  //consultar PreTomaGeneral
		{
			try
			{
				List<Models.PreTomaGeneralDetalle> result;
					var temp = from c in entities.PreTomaGeneralDetalles
							   select c;
					result = temp.ToList<Models.PreTomaGeneralDetalle>();
				
				if (result.Count > 0)
				{
					return result;
				}
				else
				{
					return result = null;
				}
			}
			catch (Exception ex)
			{
				throw ex;
			}
		}

		public int Editar(decimal id, Models.PreTomaGeneral nuevo) //editar PreTomaGeneral
		{
			try
			{
				var p = entities.PreTomaGenerals.Find(id);
				Models.PreTomaGeneral viejo = p;
				if (viejo != null)
				{
					viejo.IdPreToma = nuevo.IdPreToma;
					viejo.Fecha = nuevo.Fecha;
					viejo.Anulado = nuevo.Anulado;
					viejo.Aplicado = nuevo.Aplicado;
					viejo.Encargado = nuevo.Encargado;

					entities.Entry(viejo).State = EntityState.Modified;

					return entities.SaveChanges();
				}
				else
				{
					return 0;// no se encotro el registro solicitado.
				}
			}
			catch (Exception ex)
			{
				throw ex;
			}
		}

		public int Borrar(decimal id) //borrar PreTomaGeneral
		{
			try
			{
				var p = entities.PreTomaGenerals.Find(id);
				entities.Remove(p);
				return entities.SaveChanges();
			}
			catch (Exception ex)
			{
				throw ex;
			}
		}
	}
}


